File: play-core/src/main/kotlin/play/util/collection/ConcurrentObjectLongMap.java
package play.util.collection;

import play.util.function.ObjLongFunction;

import javax.annotation.Nullable;
import java.util.function.BiFunction;
import java.util.function.ToLongFunction;

public interface ConcurrentObjectLongMap<K> extends Iterable<ConcurrentObjectLongMap.Entry<K>> {

    Long get(K key);

    long getOrDefault(K key, long defaultValue);

    Long put(K key, long value);

    @Nullable
    Long putIfAbsent(K key, long value);

    @Nullable
    Long remove(K key);

    boolean remove(K key, long value);

    Long computeIfPresent(K key, ObjLongFunction<K, Long> remappingFunction);

    Long computeIfAbsent(K key, ToLongFunction<K> function);

    Long compute(K key, BiFunction<K, Long, Long> remappingFunction);

    boolean containsKey(K key);

    boolean isEmpty();

    boolean isNotEmpty();

    int size();

    interface Entry<K> {
        Long getKey();

        long getValue();
    }
}

File: play-core/src/main/kotlin/play/util/collection/ConcurrentObjectLongMap.kt
package play.util.collection

import play.util.function.ObjLongToObjFunction
import play.util.function.ObjToLongFunction
import java.util.function.BiFunction
import java.util.function.Function
import java.util.function.LongBinaryOperator

interface ConcurrentObjectLongMap<K> : Iterable<ConcurrentObjectLongMap.Entry<K>> {
  operator fun get(key: K): Long?
  operator fun set(key: K, value: Long): Long? = put(key, value)
  fun getOrDefault(key: K, defaultValue: Long): Long
  fun put(key: K, value: Long): Long?
  fun putIfAbsent(key: K, value: Long): Long?
  fun remove(key: K): Long?
  fun remove(key: K, value: Long): Boolean
  fun computeIfPresent(key: K, remappingFunction: ObjLongToObjFunction<K, Long>): Long?
  fun computeIfAbsent(key: K, function: Function<K, Long?>): Long?
  fun computeIfAbsent(key: K, function: ObjToLongFunction<K>): Long
  fun compute(key: K, remappingFunction: BiFunction<in K, Long?, Long?>): Long?
  fun replace(key: K, oldValue: Long, newValue: Long): Boolean
  fun merge(key: K, value: Long, remappingFunction: LongBinaryOperator): Long
  fun containsKey(key: K): Boolean
  val isEmpty: Boolean
  val isNotEmpty: Boolean
  val size: Int
  val keys: Iterable<K>
  val values: LongIterable
  fun clear()
  interface Entry<K> {
    val key: K
    val value: Long
    operator fun component1() = key
    operator fun component2() = value
  }

  companion object {
    @JvmName("create")
    operator fun <K> invoke(): ConcurrentObjectLongMap<K> = ConcurrentObjectLongHashMap()

    @JvmName("create")
    operator fun <K> invoke(initialSize: Int): ConcurrentObjectLongMap<K> = ConcurrentObjectLongHashMap(initialSize)
  }
}
